mat/io.go

```diff
@@ -29,22 +32,27 @@ var (
 // MarshalBinary encodes the receiver into a binary form and returns the result.
 //
 // Dense is little-endian encoded as follows:
-//   0 -  7  number of rows    (int64)
-//   8 - 15  number of columns (int64)
-//  16 - ..  matrix data elements (float64)
+//   0 -  1  'D', 'G' - type annotation (int8)
```

Comment:
We should probably document somewhere what these type annotations are and their complete list.

Reply:
Also, their expected order for the Dense case.

Reply:
Yeah, there isn't a complete list yet. I should probably go through blas64 and devise one.

Note that the dense case does explain the order, the 'D' and 'G' are not members of a set.